core/src/test/java/com/linecorp/armeria/common/stream/HttpDeframerTest.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.stream;
+
+import java.nio.charset.StandardCharsets;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.ResponseHeaders;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+class HttpDeframerTest {
+
+    @Test
+    void insufficientData() {
```

Comment:
Can we have test cases for N < M and N > M(if there already exists could you add a little comment or rename to recognize easily?)
Also, can we verify that the downstream subscriber actually gets what it demands by counting one by one in those test cases?

Reply:
> Can we have test cases for N < M and N > M(if there already exists could you add a little comment or rename to recognize easily?)

It was tested already. 😅 Let me clarify it.


> Also, can we verify that the downstream subscriber actually gets what it demands by counting one by one in those test cases?

The backpressure would be covered by `HttpStreamReaderTckTest`.
https://github.com/line/armeria/blob/fb091d30f5ad3b189f185bc744911dc59c176634/grpc/src/test/java/com/linecorp/armeria/internal/common/grpc/HttpStreamReaderTckTest.java#L41 

However, it looks good to add verification for `HttpDeframer` itself.